### ISSUE TITLE
Separate IDs for certain method group to delegate conversions

### ISSUE THREAD

**Issue by sharwell** [Microsoft/RoslynClrHeapAllocationAnalyzer#32]
The following conversions from method group to delegate should not be reported using the **HeapAnalyzerMethodGroupAllocationRule** ID, since these are the patterns which are used to *resolve* such an allocation:

1. A static field initializer:

    ```csharp
    private static readonly Func<string, bool> s_fileExists = File.Exists;
    ```

2. Assignment to a static field in a static constructor:

    ```csharp
    private static readonly Func<string, bool> s_fileExists;

    static MyType()
    {
        s_fileExists = File.Exists;
    }
    ```

3. Assignment to an instance field in a non-static constructor, when the method group resolves to a non-static method:

    ```csharp
    private readonly Func<string, bool> _fileExists;

    MyType()
    {
        _fileExists = FileExists;
    }
    ```

**Comment by mjsabby** [comment]
@sharwell reasonable. PRs welcome.

**Comment by edespong** [comment]
2 and 3 are not giving me any reports. Perhaps the change in 9642781 that updated the Roslyn version changed the behavior?

This is still relevant for 1, though.

**Comment by mjsabby** [comment]
Fixed in #39
